# Project Cycle 2 - 03 inference
## 1. Load Processed Data

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

sad_df = pd.read_csv("../data/processed/sad_cleaned.csv")
bmi_df = pd.read_csv("../data/processed/bmi_cleaned.csv")

## 2. Proportion Analysis (SadOrHopeless)
### 2-1 Sample Proportion

In [2]:
p_hat = sad_df["Sad_binary"].mean()
n = len(sad_df)

p_hat, n

(0.29996388587937883, 13845)

The sample proportion of students who feel sad or hopeless is approximately 0.30 (p̂ ≈ 0.29996) based on a sample size of 13,845. This estimate is very close to the benchmark value (p0 = 0.30), suggesting that there may not be a substantial difference.

### 2-2 Confidence Interval for Proportion

In [3]:
z = 1.96
se = np.sqrt(p_hat * (1 - p_hat) / n)

ci_lower = p_hat - z * se
ci_upper = p_hat + z * se

(ci_lower, ci_upper)

(0.2923307270442763, 0.30759704471448135)

The 95% confidence interval for the population proportion is approximately (0.2923, 0.3076). Since the benchmark value (p0 = 0.30) lies within this interval, there is no strong evidence that the true proportion differs from 0.30.

### 2-3 One-Sample Proportion Test

In [4]:
p0 = 0.30

z_stat = (p_hat - p0) / se
p_value = 2 * (1 - stats.norm.cdf(abs(z_stat)))

z_stat, p_value

(-0.009273182694948381, 0.9926011767381813)

The p-value is approximately 0.9926, which is much greater than the significance level (α = 0.05). Therefore, we fail to reject the null hypothesis. There is no sufficient evidence to conclude that the true proportion of students who feel sad or hopeless is different from 0.30.

## 3. Mean Analysis of BMIPCT
### 3-1 Sample Mean

In [5]:
x_bar = bmi_df["BMIPCT"].mean()
s = bmi_df["BMIPCT"].std()
n_bmi = len(bmi_df)

x_bar, s, n_bmi

(64.82068317411304, 27.516756256180642, 13062)

The sample mean BMI percentile is approximately 64.82 based on a sample size of 13,062. This value is slightly below the benchmark value (μ0 = 65), suggesting a small difference that will be further tested in the inference stage.

### 3-2 Confidence Interval for Mean

In [6]:
t_crit = stats.t.ppf(0.975, df=n_bmi-1)
se_mean = s / np.sqrt(n_bmi)

ci_lower = x_bar - t_crit * se_mean
ci_upper = x_bar + t_crit * se_mean

(ci_lower, ci_upper)

(64.34874975076963, 65.29261659745646)

The 95% confidence interval for the mean BMI percentile is approximately (64.35, 65.29). Since the benchmark value (μ0 = 65) lies within this interval, there is no strong evidence that the true mean differs from 65.

### 3-3 One-Sample t-test

In [7]:
mu0 = 65

t_stat = (x_bar - mu0) / se_mean
p_value = 2 * (1 - stats.t.cdf(abs(t_stat), df=n_bmi-1))

t_stat, p_value

(-0.7447810972515212, 0.45641746445840203)

The t-test gives a p-value of approximately 0.4564, which is greater than the significance level (α = 0.05). Therefore, we fail to reject the null hypothesis. There is no sufficient evidence to conclude that the true mean BMI percentile is different from 65.

## 4. Final Interpretation

### Conclusion
For the proportion analysis, the sample proportion of students who feel sad or hopeless is approximately 0.30. The 95% confidence interval includes the benchmark value of 0.30, and the p-value from the one-sample proportion test is much greater than 0.05. Therefore, we fail to reject the null hypothesis and conclude that there is no sufficient evidence that the true proportion differs from 0.30.

For the mean analysis, the sample mean BMI percentile is approximately 64.82, which is slightly below the benchmark value of 65. The 95% confidence interval includes 65, and the p-value from the one-sample t-test is 0.4564, which is also greater than 0.05. Therefore, we fail to reject the null hypothesis and conclude that there is no sufficient evidence that the true mean BMI percentile differs from 65.

Overall, the inferential results are consistent with the patterns observed in the EDA. The proportion of students who feel sad or hopeless appears very close to the benchmark value, and the average BMI percentile is also close to 65. In addition, the extra EDA suggests a slight increase in the proportion of sadness across BMI groups, but this pattern should be interpreted cautiously because the current inference analysis did not directly test the association between BMI group and sadness.